recipes/moc/all/conanfile.py

```diff
@@ -14,12 +13,17 @@ class MocConan(ConanFile):
     license = "Apache-2.0"    
     generators = "cmake", "cmake_find_package"
     settings = "os", "arch", "compiler", "build_type"
+    settings = {"os": ["Linux", "Macos"],
```

Comment:
These restrictions need to be raised in the `validate()` method, we never limit setting this way

Reply:
I've removed this and moved the checks in the configure method to the validate method, but FYI these recipes android-ndk, nodejs, directshowbaseclasses, vaapi, vdpau, wtl do set up the settings property as I did. They're where I discovered the syntax.

Reply:
Those are exceptions not the rule but yes my use of "never" was over zealous